CONTRIBUTING.adoc

```diff
@@ -193,6 +193,103 @@ Follow these steps to ensure that virtual box interfaces are unmanaged:
 
     $ sudo nmcli con reload
 
+=== Develop and test using a docker-in-docker cluster
+
+It's possible to run an OpenShift multinode cluster on a single host
+via docker-in-docker (dind).  Cluster creation is cheaper since each
+node is a container instead of a VM.  This was implemented primarily
+to support multinode network testing, but may prove useful for other
+use cases.
+
+To run a dind cluster in a VM, follow steps 1-3 of the Vagrant
+instructions and then execute the following:
+
+        $ export OPENSHIFT_DIND_DEV_CLUSTER=true
+        $ vagrant up
+        $ vagrant ssh
+        $ cd /vagrant
+        $ hack/dind-cluster.sh start
+
+These steps will create a new VM, update its packages, ensure that
+docker is installed, and launch a new OpenShift dind cluster.  Package
+installation and the building of the dind image will take some time,
+but are a one-time cost.  Subsequent cluster launches should be
+considerably faster.
```

Comment:
"dind-cluster.sh start" creates containers that mount the repo, builds the binaries and configuration in the master container as part of the master deploy, and then deploys to the node containers.  It's very similar to how the vagrant deploy works (equivalent to 'vagrant up').  The 'one-time cost' I'm referring to is the package installation on the vm (dnf update && dnf install docker...) and the base dind image (dnf update && dnf install openvswitch docker...).

Would it be useful to make that explicit?  e.g. "Package installation on the VM and building the dind image" and "openshift will be built as part of cluster creation"?